cmds/bmp/sl_command.go

```diff
@@ -45,10 +50,54 @@ func (cmd slCommand) Options() cmds.Options {
 
 func (cmd slCommand) Validate() (bool, error) {
 	cmd.printer.Printf("Validating %s command: options: %#v", cmd.Name(), cmd.options)
+
+	if !cmd.options.Packages && cmd.options.PackageOptions == "" {
+		return false, errors.New("Please specify --packages or --package-options")
+	}
+
 	return true, nil
 }
 
 func (cmd slCommand) Execute(args []string) (int, error) {
 	cmd.printer.Printf("Executing %s comamnd: args: %#v, options: %#v", cmd.Name(), args, cmd.options)
-	return 0, nil
+
+	validate, err := cmd.Validate()
+	if validate == false && err != nil {
+		return 1, errors.New("bmp sl validation error")
+	}
+
+	if cmd.options.Packages {
+		slPackagesResponse, err := cmd.bmpClient.SlPackages()
+		if err != nil {
+			return 1, err
+		}
+
+		if slPackagesResponse.Status != 200 {
+			return slPackagesResponse.Status, nil
+		}
+
+		table := tablewriter.NewWriter(os.Stdout)
+		table.SetHeader([]string{"Package ID", "Name"})
+
+		length := len(slPackagesResponse.Data.Packages)
+		content := make([][]string, length)
+		for i, slPackage := range slPackagesResponse.Data.Packages {
+			content[i] = []string{
+				strconv.Itoa(slPackage.Id),
+				slPackage.Name}
+		}
+
+		for _, value := range content {
+			table.Append(value)
+		}
+
+		cmd.ui.PrintTable(table)
+		cmd.ui.Println("Packages total:", length)
+
+		return 0, nil
+	} else {
+		//TODO: Add sl --package-options
```

Comment:
Where is this code? I cannot merge partial code. Also, I strongly suggest you create two private methods passing the necessary data so that the `if ... else ...` just calls each method. Otherwise, this method will be a mess.